open_spiel/python/mfg/algorithms/distribution.py

```diff
@@ -65,16 +65,20 @@ def evaluate(self):
 
     while not self.is_terminal_from_states(listing_states):
       new_listing_states = []
+      new_distribution_over_states = collections.defaultdict(float)
```

Comment:
With this new data structure for the new distribution, self.distribution_over_states no longer needs to be a default dict (which will also be safer to ensure that we don't ask for states that are not tracked, and magically get a proba of 0).